### ISSUE TITLE
ends_at set to null after subscription is cancelled

### ISSUE THREAD

**Issue by papa28x4** [digikraaft/laravel-paystack-subscription#2]
Hi, thanks for this package.

When I first installed on monday this was the behaviour. A user cancels his subscription the **paystack_status** and **ends_at** fields remain unchanged. This is the desired behaviour because a user might still have an active (current) subscription even though he is choosing to opt out of future ones. However, when I tried again yesterday the behaviour had changed. A cancellation now wipes of **ends_at** date and sets it to null. The **paystack_status** changes for active to cancelled. Not sure what happened.

Also, when I try to restart the subscription, it doesn't restart

**Comment by timoladoyinbo** [comment]
Hello, thanks for using our package. 

I haven't been able to replicate this behaviour just yet. I will further look into this. 

Let me explain what happens when a user cancels. Just as it is stated [here](https://github.com/digikraaft/laravel-paystack-subscription#cancelling-subscriptions) when a user cancels, the `paystack_status` field is set to `complete` (as per Paystack's implementation). Now, this field is not hardcoded but the value is set to the new status of the subscription that is returned after calling Paystack's cancel endpoint ((i.e using the `cancel()` method of the subscription instance). This also applies to the `ends_at` but this field's value is changed to the `next_payment_date` attribute of the returned subscription.

A possible explanation to why you may be experiencing this behaviour is if the status of the subscription returned after calling Paystack's cancel endpoint (i.e using the `cancel()` method of the subscription instance) remains unchanged which is highly unlikely.

The same thing happens when restarting a subscription. Note that you can only restart a subscription that is cancelled.

I hope this clarifies things a bit?

I will still look into this within the next couple of days.

Regards

**Comment by papa28x4** [comment]
Thanks for prompt feedback.

But I'm surprised you couldn't replicate the error because I just did with your [demo app](https://github.com/digikraaft/laravel-paystack-subscription-demo).

See video: https://drive.google.com/file/d/1k6_omJVYqIviYk3KFOhVs4qfFsB-bFoZ/view

**Comment by timoladoyinbo** [comment]
Thanks, @papa28x4. We'll have a look at this within the next couple of days and revert.

**Comment by timoladoyinbo** [comment]
Apologies for getting back late. We've discovered what the issue is as well as some other issues we're investigating. 

We will make some updates and tag a new release within the next couple of days.

Thanks

**Comment by papa28x4** [comment]
Expectant, thank you very much @timoladoyinbo

**Comment by timoladoyinbo** [comment]
Hello, @papa28x4 we've made a new release that fixes the issue. 

The issue was caused by a change in how paystack records the subscription status.

We also noticed that when a subscription is cancelled, it cannot be restarted (enabled) any more. So at the moment, if you use the enable (`$user->subscription('default')->enable()`) method after cancelling a subscription, Paystack API will return a not found subscription exception.

Regards.
